fuse/opcode.go

```diff
@@ -450,7 +450,21 @@ func doStatFs(server *Server, req *request) {
 }
 
 func doIoctl(server *Server, req *request) {
-	req.status = ENOSYS
+	var buf []byte
+	in := (*IoctlIn)(req.inData)
+	out := (*_IoctlOut)(req.outData())
+	if in.InSize > 0 {
+		buf = req.arg
+	} else if in.OutSize > 0 {
+		outputsize := unsafe.Sizeof(_IoctlOut{})
+		if uintptr(in.OutSize) <= outputHeaderSize-sizeOfOutHeader-outputsize {
+			buf = req.outBuf[sizeOfOutHeader+outputsize : sizeOfOutHeader+outputsize+uintptr(in.OutSize)]
+		} else {
+			buf = server.allocOut(req, in.OutSize)
+		}
+		req.flatData = buf
+	}
+	out.Result = int32(server.fileSystem.Ioctl(req.cancel, in, buf))
```

Comment:
Better to make a more general `ioctl` interface here, passing in & out & buffer as arguments, and leave details to the real file system.